diff --git a/ffc.ffc_conditions_info.inc b/ffc.ffc_conditions_info.inc
index 0ee93c3..f7ea2a8 100644
--- a/ffc.ffc_conditions_info.inc
+++ b/ffc.ffc_conditions_info.inc
@@ -60,51 +60,91 @@ function _ffc_field_formatter_settings_form_alter(&$settings_form, $context, $co
     $condition_options[$key] = $info['title'];
   }
 
-  // We don't iterate for now, this is to make sure we can have
-  // multiple conditions in the future if we really need this.
-  $settings_form['conditions'][0]['condition'] = array(
-    '#type' => 'select',
-    '#empty_option' => t('- Select a condition -'),
-    '#options' => $condition_options,
-    '#default_value' => isset($settings['conditions'][0]['condition']) ? $settings['conditions'][0]['condition'] : '',
-    '#ajax' => array(
-      '#submit' => array('ffc_field_ui_display_overview_multistep_submit'),
-      'callback' => 'ffc_field_ui_display_overview_multistep_js',
-      'wrapper' => 'field-display-overview-wrapper',
-      'ffc_context' => $ffc_context,
-    ),
-  );
-
-  // Get the form.
-  if (isset($context['form_state']['values']['fields'][$field_name]['settings_edit_form']['settings']['conditions'][0]['condition'])) {
-    $condition = $context['form_state']['values']['fields'][$field_name]['settings_edit_form']['settings']['conditions'][0]['condition'];
+  // Check for existing conditions
+  if (isset($settings['conditions'][0]['condition'])) {
+    // Track which conditions have been set
+    $used_condition_items = array();
+    // Iterate conditions
+    foreach ($settings['conditions'] as $condition_key => $condition_item) {
+      // Add the condition to the list of used condition items
+      $used_condition_items[] = $condition_item['condition'];
+      // Add the condition to the settings form
+      $settings_form['conditions'][] = array(
+        'condition' => array(
+          '#type' => 'select',
+          '#empty_option' => t('- Remove this condition -'),
+          '#options' => array($condition_item['condition'] => $condition_options[$condition_item['condition']]),
+          '#default_value' => $condition_item['condition'],
+          '#ajax' => array(
+            '#submit' => array('ffc_field_ui_display_overview_multistep_submit'),
+            'callback' => 'ffc_field_ui_display_overview_multistep_js',
+            'wrapper' => 'field-display-overview-wrapper',
+            'ffc_context' => $ffc_context,
+          ),
+        ),
+      );
+      // Update available condition options
+      unset($condition_options[$condition_item['condition']]);
+    }
   }
-  elseif (!empty($settings['conditions'][0]['condition'])) {
-    $condition = $settings['conditions'][0]['condition'];
+  // If there are any unused conditions, add a new option to the form
+  if ($condition_options) {
+    // Add a new condition option
+    $settings_form['conditions'][] = array(
+      'condition' => array(
+        '#type' => 'select',
+        '#empty_option' => t('- Select a condition -'),
+        '#options' => $condition_options,
+        '#default_value' => '',
+        '#ajax' => array(
+          '#submit' => array('ffc_field_ui_display_overview_multistep_submit'),
+          'callback' => 'ffc_field_ui_display_overview_multistep_js',
+          'wrapper' => 'field-display-overview-wrapper',
+          'ffc_context' => $ffc_context,
+        ),
+      ),
+    );
   }
 
-  $open_field_set = FALSE;
-  if (!empty($condition)) {
-    $callback = 'ffc_condition_form_' . $condition;
-    if (function_exists($callback)) {
-      $open_field_set = TRUE;
-      $configuration = !empty($settings['conditions'][0]['configuration']) ? $settings['conditions'][0]['configuration'] : array();
-      $settings_form['conditions'][0]['configuration'] = $callback($ffc_context, $configuration);
+  // Iterate conditions, checking both form_state and settings
+  $condition_iterator = 0;
+  $conditions_exist = TRUE;
+  do {
+    // Get the form.
+    if (isset($context['form_state']['values']['fields'][$field_name]['settings_edit_form']['settings']['conditions'][$condition_iterator]['condition'])) {
+      $condition = $context['form_state']['values']['fields'][$field_name]['settings_edit_form']['settings']['conditions'][$condition_iterator]['condition'];
     }
-    elseif (!empty($conditions[$condition]['no form'])) {
-      $open_field_set = TRUE;
-      $settings_form['conditions'][0]['configuration'] = '';
+    elseif (!empty($settings['conditions'][$condition_iterator]['condition'])) {
+      $condition = $settings['conditions'][$condition_iterator]['condition'];
     }
     else {
-      $settings_form['conditions'][0]['error'] = array(
-        '#markup' => t('The "%callback" function was not found.', array('%callback' => $callback))
-      );
+      $condition = null;
+      $conditions_exist = FALSE;
+    }
+    $open_field_set = FALSE;
+    if (!empty($condition)) {
+      $callback = 'ffc_condition_form_' . $condition;
+      if (function_exists($callback)) {
+        $open_field_set = TRUE;
+        $configuration = !empty($settings['conditions'][$condition_iterator]['configuration']) ? $settings['conditions'][$condition_iterator]['configuration'] : array();
+        $settings_form['conditions'][$condition_iterator]['configuration'] = $callback($ffc_context, $configuration);
+      }
+      elseif (!empty($conditions[$condition]['no form'])) {
+        $open_field_set = TRUE;
+        $settings_form['conditions'][$condition_iterator]['configuration'] = '';
+      }
+      else {
+        $settings_form['conditions'][$condition_iterator]['error'] = array(
+          '#markup' => t('The "%callback" function was not found.', array('%callback' => $callback))
+        );
+      }
     }
-  }
 
-  if ($open_field_set) {
-    $settings_form['conditions']['#collapsed'] = FALSE;
-  }
+    if ($open_field_set) {
+      $settings_form['conditions']['#collapsed'] = FALSE;
+    }
+    ++$condition_iterator;
+  } while($conditions_exist);
 }
 
 /**
diff --git a/ffc.module b/ffc.module
index 206e761..106a48b 100644
--- a/ffc.module
+++ b/ffc.module
@@ -28,9 +28,13 @@ function ffc_form_field_ui_display_overview_form_alter(&$form, &$form_state) {
 function ffc_validate_field_display_overview(&$form, &$form_state) {
   if (!empty($form_state['formatter_settings']) && is_array($form_state['formatter_settings'])) {
     foreach ($form_state['formatter_settings'] as $key => $info) {
-      if (empty($info['conditions'][0]['condition'])) {
-        $form_state['formatter_settings'][$key]['conditions'] = array();
+      foreach ($info['conditions'] as $condition_key => $condition_item) {
+        if (empty($condition_item['condition'])) {
+          unset($form_state['formatter_settings'][$key]['conditions'][$condition_key]);
+        }
       }
+      // re-index the array to normalize integer keys
+      $form_state['formatter_settings'][$key]['conditions'] = array_values($form_state['formatter_settings'][$key]['conditions']);
     }
   }
 }
