diff --git a/imagezoom.module b/imagezoom.module
index cfb79bd..6a72df7 100644
--- a/imagezoom.module
+++ b/imagezoom.module
@@ -178,11 +178,8 @@ function template_preprocess_imagezoom_image(&$variables) {
   $item = $variables['item'];
 
   if ($variables['display_style']) {
-    $image_style = image_style_load($variables['display_style']);
-    $derivative_uri = image_style_path($variables['display_style'], $item['uri']);
-    image_style_create_derivative($image_style, $item['uri'], $derivative_uri);
     $variables['image'] = image_style_url($variables['display_style'], $item['uri']);
-    $info = image_get_info($derivative_uri);
+    $info = image_get_info($variables['image']);
   }
   else {
     $variables['image'] = file_create_url($item['uri']);
@@ -195,9 +192,6 @@ function template_preprocess_imagezoom_image(&$variables) {
   $variables['title'] = $item['title'];
 
   if ($variables['zoom_style']) {
-    $image_style = image_style_load($variables['zoom_style']);
-    $derivative_uri = image_style_path($variables['zoom_style'], $item['uri']);
-    image_style_create_derivative($image_style, $item['uri'], $derivative_uri);
     $variables['zoom'] = image_style_url($variables['zoom_style'], $item['uri']);
   }
   else {
diff --git a/modules/imagezoom_gallery/imagezoom_gallery.module b/modules/imagezoom_gallery/imagezoom_gallery.module
index 66d80c3..582bd5a 100644
--- a/modules/imagezoom_gallery/imagezoom_gallery.module
+++ b/modules/imagezoom_gallery/imagezoom_gallery.module
@@ -161,21 +161,13 @@ function template_preprocess_imagezoom_thumb(&$variables) {
   $item = $variables['item'];
 
   if ($variables['display_style']) {
-    $image_style = image_style_load($variables['display_style']);
-    $derivative_uri = image_style_path($variables['display_style'], $item['uri']);
-    image_style_create_derivative($image_style, $item['uri'], $derivative_uri);
     $variables['image'] = image_style_url($variables['display_style'], $item['uri']);
-    $info = image_get_info($derivative_uri);
   }
   else {
     $variables['image'] = file_create_url($item['uri']);
-    $info = image_get_info($item['uri']);
   }
 
   if ($variables['zoom_style']) {
-    $image_style = image_style_load($variables['zoom_style']);
-    $derivative_uri = image_style_path($variables['zoom_style'], $item['uri']);
-    image_style_create_derivative($image_style, $item['uri'], $derivative_uri);
     $variables['zoom'] = image_style_url($variables['zoom_style'], $item['uri']);
   }
   else {
@@ -183,11 +175,8 @@ function template_preprocess_imagezoom_thumb(&$variables) {
   }
 
   if ($variables['thumb_style']) {
-    $image_style = image_style_load($variables['thumb_style']);
-    $derivative_uri = image_style_path($variables['thumb_style'], $item['uri']);
-    image_style_create_derivative($image_style, $item['uri'], $derivative_uri);
     $variables['thumb'] = image_style_url($variables['thumb_style'], $item['uri']);
-    $info = image_get_info($derivative_uri);
+    $info = image_get_info($variables['thumb']);
   }
   else {
     $variables['thumb'] = file_create_url($item['uri']);
