From b79bbb97dbc0e9bd0fa20c17bfaf62dc8f9bf7b7 Mon Sep 17 00:00:00 2001
From: klausi
Date: Fri, 15 Jan 2016 12:04:37 +0100
Subject: Issue #2406863 by klausi: "Cache rebuild lock hit" watchdog message,
 possible deadlock

---
 rules.module | 1 -
 1 file changed, 1 deletion(-)

diff --git a/rules.module b/rules.module
index 719852c..e148163 100644
--- a/rules.module
+++ b/rules.module
@@ -366,7 +366,6 @@ function &rules_get_cache($cid = 'data') {
       // Prevent stampeding by ensuring the cache is rebuilt just once at the
       // same time.
       while (!lock_acquire(__FUNCTION__ . $cid . $cid_suffix, 60)) {
-        rules_log('Cache rebuild lock hit: !cid', array('!cid' => $cid), RulesLog::WARN);
         // Now wait until the lock is released.
         lock_wait(__FUNCTION__ . $cid . $cid_suffix, 10);
         // If the lock is released it's likely the cache was rebuild. Thus check
-- 
cgit v0.12

