diff --git a/plugins/views_plugin_style.inc b/plugins/views_plugin_style.inc
index d355e1b..99594bc 100644
--- a/plugins/views_plugin_style.inc
+++ b/plugins/views_plugin_style.inc
@@ -129,7 +129,7 @@ class views_plugin_style extends views_plugin {
 
       $classes = explode(' ', $class);
       foreach ($classes as &$class) {
-        $class = drupal_clean_css_identifier($class);
+        $class = views_clean_css_identifier($class);
       }
       return implode(' ', $classes);
     }
