diff --git a/.htaccess b/.htaccess
index 4654035..795ef6b 100644
--- a/.htaccess
+++ b/.htaccess
@@ -152,3 +152,41 @@ DirectoryIndex index.php index.html index.htm
   # Disable content sniffing, since it's an attack vector.
   Header always set X-Content-Type-Options nosniff
 </IfModule>
+
+<FilesMatch "^(css|js)__[A-Za-z0-9-_]{43}__[A-Za-z0-9-_]{43}__[A-Za-z0-9-_]{43}.(css|js)(\.gz|\.br)?">
+  # No mod_headers. Apache module headers is not enabled.
+  <IfModule !mod_headers.c>
+    # No mod_expires. Apache module expires is not enabled.
+    <IfModule !mod_expires.c>
+      # Use ETags.
+      FileETag MTime Size
+    </IfModule>
+  </IfModule>
+
+  # Use Expires Directive if apache module expires is enabled.
+  <IfModule mod_expires.c>
+    # Enable expirations.
+    ExpiresActive On
+    # Cache all aggregated css/js files for 52 weeks after access (A).
+    ExpiresDefault A31449600
+  </IfModule>
+
+  # Use Headers Directive if apache module headers is enabled.
+  <IfModule mod_headers.c>
+    <IfModule !mod_expires.c>
+      # Set a far future Cache-Control header to 52 weeks.
+      Header set Cache-Control "max-age=31449600, no-transform, public"
+    </IfModule>
+    <IfModule mod_expires.c>
+      Header append Cache-Control "no-transform, public"
+    </IfModule>
+  </IfModule>
+</FilesMatch>
+# Force advagg .js file to have the type of application/javascript.
+<FilesMatch "^js__[A-Za-z0-9-_]{43}__[A-Za-z0-9-_]{43}__[A-Za-z0-9-_]{43}.js(\.gz|\.br)?">
+  ForceType application/javascript
+</FilesMatch>
