diff --git a/.htaccess b/.htaccess
index 440cabc..2aa3247 100644
--- a/.htaccess
+++ b/.htaccess
@@ -116,6 +116,11 @@ DirectoryIndex index.php index.html index.htm
   RewriteCond %{REQUEST_URI} !=/favicon.ico
   RewriteRule ^ index.php [L]
 
+  # Provide a redirect for the FeedBurner feed.
+  RewriteEngine on
+  RewriteCond %{HTTP_USER_AGENT} !FeedBurner
+  RewriteRule ^rss\.xml$ https://feeds.feedburner.com/titaniumrings [R,L]
+
   # Rules to correctly serve gzip compressed CSS and JS files.
   # Requires both mod_rewrite and mod_headers to be enabled.
   <IfModule mod_headers.c>
